src/inputs/DescriptionEditor/DescriptionEditor.js

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import EditIcon from '@mui/icons-material/Edit';
+import { Grid, Stack, TextField, Typography } from '@mui/material';
+
+export const DescriptionEditor = (props) => {
+  const { onChange, readOnly, value, scenarioId, labels } = props;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+  const [descriptionText, setDescriptionText] = useState(value);
+
+  useEffect(() => {
+    if (value !== descriptionText) setDescriptionText(value);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value]);
+
+  const startEdition = useCallback((event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setIsEditing(true);
+  }, []);
+
+  const stopEdition = useCallback(
+    (event) => {
+      event.stopPropagation();
+      event.preventDefault();
+      setIsEditing(false);
+
+      if (onChange && event.target.value !== value) {
+        onChange(event.target.value);
```

Comment:
`event.target.value` is used twice so we can set it in a const.

Reply:
I'd say both versions are ok :smile:

Reply:
I don't have a strong opinion but `event.target.value` always seems easy to read :sweat_smile: